### PR TITLE
[FEATURE] show subtasks as indented in checklist view

### LINKED ISSUE #25: Sub tasks indentation
I read through the opened and closed issues, and saw you reference the problem of sub tasks not being indented in the task list, and even read one where you imply that you are looking for a way to implement it.
I just wanted to know if that was still in your plans.
Thank you for your time.

### PR DESCRIPTION
**Pull request by GamerGirlandCo**
closes #25

### CODE REVIEW

File: src/utils/tasks.ts [hunk 1]
```diff
@@ -49,142 +86,44 @@ export const parseTodos = async (
...
-    if (line.length === 0) break
-    if (lineIsValidTodo(line)) {
-      todos.push(formTodo(line, file, links, i, tagMeta))
+  ): Promise<Map<TFile, TodoItem[]>> => {
+    const includePattern = includeFiles.trim()
+    ? includeFiles.trim().split("\n")
+    : ["**/*"];
+    
+    const todosForUpdatedFiles = new Map<TFile, TodoItem[]>();
+    const query = `(${includeFiles.split("\n").map(a => '"' + (a || "/") + '"').join(" or ")})${todoTags.filter(a => a.trim() != "*").length ? " and (" + todoTags.map(a => "#" + a).join(" ") + ")" : ""}`
```
Comment (GamerGirlandCo) [inline src/utils/tasks.ts:95]: here is where dataview is used

Comment (GamerGirlandCo) [inline src/utils/tasks.ts:95]: this constructs a dataview query based on the user's filter settings

File: src/utils/tasks.ts [hunk 2]
```diff
@@ -49,142 +86,44 @@ export const parseTodos = async (
...
+    for (const page of dv.pages(query)) {
```
Comment (GamerGirlandCo) [inline src/utils/tasks.ts:96]: here is how dataview is being used, with the pages query constructed above.

File: src/utils/tasks.ts [hunk 3]
```diff
@@ -21,25 +22,61 @@ import {
...
+  TagCache,
+  TFile,
+  Vault,
+} from "obsidian";
+import type { TodoItem, TagMeta, FileInfo, mapFn } from "src/_types";
+
+
+const md = new MD().use(commentPlugin).use(tagPlugin).use(highlightPlugin)
+// @ts-ignore it exists shhhhhhhhhhhhhhh
+const dv = window.app.plugins.plugins["dataview"].api
```
Comment (delashum) [inline src/utils/tasks.ts:40]: lol

**Comment by delashum** [comment]
Definitely not adding a dependency on another plugin :/ sorry

**Pull request closed without merging**